dvc/output/local.py

```diff
@@ -48,7 +48,11 @@ def _parse_path(self, fs, path):
         return self.FS_CLS.PATH_CLS(abs_p)
 
     def __str__(self):
-        if not self.repo or not self.is_in_repo:
+        if (
+            not self.repo
+            or urlparse(self.def_path).scheme == "remote"
+            or os.path.isabs(self.def_path)
+        ):
             return str(self.def_path)
 
         cur_dir = os.getcwd()
```

Comment:
We expect ([and have tests for it](https://github.com/iterative/dvc/blob/b2f71d459bab129833840c8c5517f28156ff6162/tests/unit/output/test_local.py#L31)) that `str()` of an external output with a relative path in `def_path` should be a proper `relpath` and not just a `def_path`, which no longer fits corrected `is_in_repo`.

Reply:
Excuse me ,what is the difference between them? `def_path` and `relpath`, one relative to current dir another to repo root?

Reply:
@karajan1001, `def_path` is what is defined (or, to-be defined) in the `dvc.yaml` and `.dvc` files. `relpath` depends on what we are using for (based on stage's wdir or root of the repo or the current working directory).

Reply:
@skshetry , so maybe `__repr__` should return `def_path` and `__str__` returns `relpath` ?

Reply:
@karajan1001, I'd expect `__repr__` to return something like `Output: {def_path}` or `Output: {resolved_path}` as it makes it easier to debug. `output.def_path` is also much better than `repr(output)`.